catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/cache/solr/impl/CachingFederationStrategyTest.java

```diff
@@ -0,0 +1,383 @@
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.cache.solr.impl;
+
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static ddf.catalog.cache.solr.impl.CachingFederationStrategy.CACHE_QUERY_MODE;
+import static ddf.catalog.cache.solr.impl.CachingFederationStrategy.INDEX_QUERY_MODE;
+import static ddf.catalog.cache.solr.impl.CachingFederationStrategy.NATIVE_QUERY_MODE;
+import static ddf.catalog.cache.solr.impl.CachingFederationStrategy.QUERY_MODE;
+import static ddf.catalog.cache.solr.impl.CachingFederationStrategy.UPDATE_QUERY_MODE;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import org.codice.ddf.itests.common.WaitCondition;
+import org.geotools.filter.NullFilterImpl;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.opengis.filter.sort.SortBy;
+
+import ddf.catalog.Constants;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.ResultImpl;
+import ddf.catalog.filter.proxy.builder.GeotoolsFilterBuilder;
+import ddf.catalog.impl.MockSource;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.operation.impl.DeleteResponseImpl;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.QueryResponseImpl;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+import ddf.catalog.operation.impl.UpdateResponseImpl;
+import ddf.catalog.plugin.PreFederatedQueryPlugin;
+import ddf.catalog.source.Source;
+
+public class CachingFederationStrategyTest {
```

Comment:
❗️ General comment on all the tests: we need to make sure we have `verify` calls with the proper argument matchers for all the dependencies the `CachingFederationStrategy` is using, otherwise the unit tests may not fail when a breaking change is introduced.